assets/src/edit-story/components/form/textArea.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { rgba } from 'polished';
+
+const StyledTextArea = styled.textarea`
+  width: 100%;
+  padding: 0;
+  border: none;
+  appearance: none;
+  box-shadow: none !important;
+  outline: none;
+  background-color: transparent;
+  color: ${({ theme }) => theme.colors.fg.white};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  resize: ${({ resizeable }) => (resizeable ? 'auto' : 'none')};
+
+  &:disabled {
+    background-color: transparent;
+    color: ${({ theme }) => theme.colors.fg.white};
+  }
+`;
+
+const Container = styled.div`
+  width: 100%;
+  color: ${({ theme }) => rgba(theme.colors.fg.white, 0.3)};
+  background-color: ${({ theme }) => rgba(theme.colors.bg.black, 0.3)};
+
+  ${({ disabled, readOnly }) => (disabled || readOnly) && `opacity: 0.3`};
+
+  padding: 6px 12px;
+  padding-right: 6px;
+  border-radius: 4px;
+  border: 1px solid transparent;
+  &:focus-within {
+    border-color: ${({ theme }) => theme.colors.whiteout} !important;
+  }
+`;
+
+const Counter = styled.div`
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.tab.size};
+  text-align: right;
+  line-height: 1;
+  padding-right: 6px;
+`;
+
+function TextArea({
+  className,
+  placeholder,
+  value,
+  maxLength,
+  readOnly,
+  disabled,
+  resizeable,
+  showTextLimit,
+  rows,
+  onTextChange,
+  onBlur,
+  ...rest
+}) {
+  const hasMaxLength = useMemo(() => typeof maxLength === 'number', [
+    maxLength,
+  ]);
```

Comment:
Super small and unimportant nit, but I think this is actually a case where `useMemo` isn't worth using since the expression isn't very expensive (probably less so so than the invocation of useMemo)

Reply:
I fully agree, my instinct was to _not_ use `useMemo` but then I wanted `hasMaxLength` to be a dependency of the `handleChange` `useCallback` implementation, so I immediately thought I had to `useMemo` `hasMaxLength` so it's reference doesn't change per re-render thus forcing `handleChange` to re-create each time too, buuuuuuut `hasMaxLength` is a bool so even if the component gets re-rendered a bunch of times, if `hasMaxLength` continually gets assigned `false`, it won't trigger `handleChange` to update because it's not comparing against the reference but against the value.

Not sure if my rambling above makes sense but, the conclusion is, I agree :D and I'll remove the `useMemo` :D